roles/validations/tasks/edpm/bmh_compute_replace.yml

```
@@ -0,0 +1,196 @@
+# This job tests the functionality of the openstack-operator to replace a
+# bmh compute node.
+#
+# This job was created to satisfy:
+# https://issues.redhat.com/browse/OSPRH-15061
+
+- name: Get name of nodeset containing bmh node to be used as spare node
+  cifmw.general.ci_script:
+    output_dir: "{{ cifmw_validations_basedir }}/artifacts"
+    script: >-
+      oc -n {{ cifmw_validations_namespace }} get bmh {{ cifmw_validations_bmh_spare_leaf_label }} --no-headers | awk '{print $3}'
+  register: bmh_nodeset_name
+
+- name: Verify bmh node being used as spare is provisioned before scale down
+  cifmw.general.ci_script:
+    output_dir: "{{ cifmw_validations_basedir }}/artifacts"
+    script: >-
+      oc -n {{ cifmw_validations_namespace }} get bmh {{ cifmw_validations_bmh_spare_leaf_label }}
+  register: bmh_nodes_before_scale_down
+  failed_when: '"provisioned" not in bmh_nodes_before_scale_down.stdout'
+
+- name: Get compute service list
+  cifmw.general.ci_script:
+    output_dir: "{{ cifmw_validations_basedir }}/artifacts"
+    script: >-
+      oc -n {{ cifmw_validations_namespace }} rsh openstackclient openstack compute service list
+  register: compute_service_list_out
+  until: '"{{ cifmw_validations_bmh_spare_hostname }}" in compute_service_list_out.stdout'
+
+- name: Disable nova-compute for node being removed
+  cifmw.general.ci_script:
+    output_dir: "{{ cifmw_validations_basedir }}/artifacts"
+    script: >-
+      oc -n {{ cifmw_validations_namespace }} rsh openstackclient openstack compute service set {{ cifmw_validations_bmh_spare_hostname }} nova-compute --disable
+  register: compute_service_set_out
+  until: '"Failed" not in compute_service_set_out.stdout'
+
+- name: Get ovn controller id of host to be removed
+  cifmw.general.ci_script:
+    output_dir: "{{ cifmw_validations_basedir }}/artifacts"
+    script: >-
+      oc -n {{ cifmw_validations_namespace }} rsh openstackclient openstack network agent list --host {{ cifmw_validations_bmh_spare_hostname }} | grep "OVN Controller agent"  | awk '{print $2}'
```

```suggestion
      oc -n {{ cifmw_validations_namespace }} rsh openstackclient openstack network agent list --host {{ cifmw_validations_bmh_spare_hostname }} -f value -c ID
```

That will just give you the value:
```
oc rsh openstackclient openstack network agent list --host edpm-compute-1.ctlplane.example.com -f value -c ID
945d481a-20c4-45d3-8c92-3b2c04ba628f
```